### DISCUSSION TITLE
redirects or rewrites with anchors/jump links

### DISCUSSION THREAD

**Discussion by Sogeman** [vercel/next.js#34428] [Help]
Hi,
is it not possible to redirect to a page with an anchor or rewrite a url to include an anchor (not even sure if the rewrite would trigger the scroll).

I tried redirecting like:

    async redirects() {
        return [
            {
                source: '/testredirect/',
                destination: '/test#redirect',
                permanent: true,
            },
        ];
    },

or rewrite it like

 async rewrites() {
   return [

            {
                source: '/test/',
                destination: '/test#rewrite',
            },
        ];
    }

but neither seems to work.
The rewrite doesn't seem to work at all and the redirect ends in some redirect error.

Should that work and am I doing it wrong? Or is it only possible by defining my own array of redirects and checking on page load and manually jumping to the anchor.

thanks

**Comment by 2stash** [comment]
@Sogeman Is there a reason you are using a "#" for your destination paths instead of just using a '/' ?

**Reply by Sogeman** [reply]
@2stash yes, we're re-launching a website and the URL structure changed. Content from, for example /test/topic1/ is now part of the new page /test/ and there's a quick access bar to jump to these section on a page.
But old links should redirect to that part of the new page directly.

**Reply by DonHuskini** [reply]
@Sogeman how did you solve it?

**Reply by Sogeman** [reply]
We didn't. We just threw out those redirects because the few pages that used it weren't worth the effort

**Comment by Richie94** [comment]
is there any update here? im facing the same problem, having some redirect "redirect('/event/1#elem-12')" which seems to get ignored and i only get redireted to /event/1...

**Comment by cdorfer** [comment]
Same here..
